src/edu/ucsd/library/xdre/tab/RDFExcelConvertor.java

```diff
@@ -0,0 +1,244 @@
+
+				String fieldValues = dataRow.get(columnName);
+				if (fieldValues == null) 
+					fieldValues = columnValue;
+				else
+					fieldValues += "|" + columnValue;
+
+				dataRow.put(columnName, fieldValues);
+			}
+		}
+
+		// update the counts of each field
+		updateFieldCounts(fieldCounts, recFieldCounts);
+	}
+
+	private void updateFieldCounts(Map<String, Integer> fieldCounts, Map<String, Integer> tmpCounts) {
+		// update the counts of each field
+		for (String fieldName : tmpCounts.keySet()) {
+			Integer count = fieldCounts.get(fieldName);
+			if (count == null || tmpCounts.get(fieldName) > fieldCounts.get(fieldName))
+				fieldCounts.put(fieldName, new Integer(tmpCounts.get(fieldName).intValue()));
+		}
+	}
+
+	private void appendValues(StringBuilder line, int fieldCount, String fieldValues) {
+		if (fieldValues != null) {
+			String[] values = fieldValues.split("\\|");
+
+			for (String value : values) {
+				if (line.length() > 1 || (line.length() == 1 && line.charAt(0) != ','))
+					line.append(",");
+
+				if (value.indexOf(",") >= 0 || value.indexOf("\"") >= 0 
+						|| value.indexOf(System.getProperty("line.separator")) >= 0)
+					line.append("\"" + value.replace("\"", "\"\"") + "\"");
+				else
+					line.append(value);
+			}
+
+			// append commas for extra fields
+			if (fieldCount > values.length) {
+				for (int i= values.length; i< fieldCount; i++) {
+					line.append(",");
+				}
+			}	
+		} else {
+			for(int i=0; i<fieldCount; i++)
+				line.append(",");
+		}
+	}
+
+	private void appendFlatColumnNames(StringBuilder line, int fieldCount, String columnName) {
+		for (int i=0; i< fieldCount; i++) {
+			if (line.length() > 1)
+				line.append(",");
+	
+			line.append(columnName);
+		}
+	}
+
+	/**Convert source to other formats with xsl transform
+	 * 
+	 * @param xsl
+	 * @param source
+	 * @return
+	 * @throws FileNotFoundException
+	 * @throws TransformerException
+	 */
+	public static String xslConvert(String xsl, String source) throws FileNotFoundException, TransformerException {
+		Reader xslReader = null;
+		InputStream srcInput = null;
+		OutputStream out = null;
+		try{
+			out = new ByteArrayOutputStream();
+			xslReader = new FileReader(xsl);
+			Source xslSrc = new StreamSource(xslReader);
+			TransformerFactory transFact = TransformerFactory.newInstance( );
+			Templates xslTemp = transFact.newTemplates(xslSrc);
+			Transformer trans = xslTemp.newTransformer();
+			srcInput = new FileInputStream(source);
+			Source xmlSource = new StreamSource(srcInput);
+			StreamResult result = new StreamResult(out);
+			trans.setOutputProperty("encoding", "UTF8");
+			trans.setOutputProperty(OutputKeys.INDENT, "yes");
+			trans.transform(xmlSource, result);
+		}finally{
+			close(xslReader);
+			close(srcInput);
+			close(out);
+		}
+		return out.toString();
+	}
+
+	public static void close(Closeable closable){
+		if(closable != null){
+			try {
+				closable.close();
+			} catch (IOException e) {}
```

Comment:
Like you said, it's an edge case that shouldn't happen and if it happens than I don't know whether we care about it. I just added the try block to avoid this kind of error break the process. Yes, it won't harm to add a line to log it.